osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -559,7 +559,11 @@ public WorkingBeatmap GetWorkingBeatmap(BeatmapInfo? beatmapInfo, bool refetch =
                 // If we seem to be missing files, now is a good time to re-fetch.
                 bool missingFiles = beatmapInfo.BeatmapSet?.Files.Count == 0;
 
-                if (refetch || beatmapInfo.IsManaged || missingFiles)
+                if (beatmapInfo.IsManaged)
+                {
+                    beatmapInfo = beatmapInfo.Detach();
+                }
+                else if (refetch || missingFiles)
```

Comment:
This change makes me nervous but I don't have any concrete reason to doubt it so I'm just gonna hope this breaks nothing